### PR TITLE
Login

### PR DESCRIPTION
**Pull request by pelikhan**
image
image

PR started by Sam with better support for OneDrive/Google drive; undusted. Cloud provider (onedrive/google drive) unified with Github support.

Github support still using dev token UI (current UI).

Test editor: https://maker.makecode.com/app/bc72ed6d982c0fe5747087e70760688b99e4b2f8-b4020ae11c

- [x] display login on home screen if not signed in
- [x] display bubble head on home screen + editor if signed in
- [x] display "New repository" menu item in bubble head (not showing "create repository in file menu anymore)
- [x] confirmation dialog for sign out
- [x] unify github/onedrive/google signin, split from cloud sync
- [x] hide "create GitHub repository" unless signed in with github
- [x] sign out delete all repoes

## non goals:

- oauth roundtrip for github
- test github
-  test onedrive
- test google drive

### CODE REVIEW

File: webapp/public/oauth-redirect.html [hunk 1]
```diff
@@ -0,0 +1,64 @@
+<!DOCTYPE html>
+<html>
+
+<body id='root' style='font-size:16px; font-family:sans-serif;'>
+    <p id="msg">
+    </p>
+    <script>
+        var id = "minecraft";
```
Comment (pelikhan) [inline webapp/public/oauth-redirect.html]: needs fix

**Comment by darzu** [comment]
Hmm:
image

**Comment by pelikhan** [comment]
@darzu the uploaded editor link is stale.

**Comment by pelikhan** [comment]
Editor link updated

File: svgicons/googledrive.svg [hunk 2]
```diff
@@ -0,0 +1,70 @@
+<?xml version="1.0" encoding="UTF-8" standalone="no"?>
+<svg
```
Comment (mmoskal) [inline svgicons/googledrive.svg]: run svgo on this?

File: webapp/src/cloudsync.ts [hunk 3]
```diff
@@ -416,37 +514,51 @@ export function syncAsync(): Promise<void> {
...
         .then(() => provider.listAsync())
         .then(entries => {
-            let allScripts = ws.getHeaders()
+            // Get all local headers including those that had been deleted
+            let allScripts = ws.getHeaders(true)
             let cloudHeaders = U.toDictionary(entries, e => e.id)
-            let existingHeaders = U.toDictionary(allScripts, h => h.blobId)
-            let waitFor = allScripts.map(hd => {
+            let existingHeaders = U.toDictionary(allScripts.filter(h => h.blobId), h => h.blobId)
+            console.log('all', allScripts);
```
Comment (mmoskal) [inline webapp/src/cloudsync.ts]: make logging optional?

Comment (pelikhan) [reply webapp/src/cloudsync.ts]: removing. note that this code path is only used by onedrive/google drive.

Comment (pelikhan) [reply webapp/src/cloudsync.ts]: svgoed

File: svgicons/googledrive.svg (hunk 2 above)
Comment (pelikhan) [reply svgicons/googledrive.svg]: svgoed

**Pull request merged**